core/evm_ext.go

```diff
@@ -0,0 +1,107 @@
+// (c) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package core
```

Comment:
Why does this live in `core` instead of `vm`? Being here blurs any existing separation of concerns.

Reply:
It cannot be in `core/vm` since the purpose is moving to a direction where avalanche packages are not imported in geth code.
So `core` is better than `core/vm`, which is more arguably the inner most geth code. We can move it to another package if desired, originally I picked `core` because it was already imported in most callsites of this code.